mdata/store_cassandra.go

```diff
@@ -514,17 +526,24 @@ func (c *CassandraStore) SearchTable(ctx context.Context, key, table string, sta
 	outcomes := make([]outcome, 0, numQueries)
 
 	seen := 0
-	for o := range results {
-		if o.omitted {
-			tracing.Failure(span)
-			tracing.Error(span, errReadTooOld)
-			return nil, errReadTooOld
-		}
-		seen += 1
-		outcomes = append(outcomes, o)
-		if seen == numQueries {
-			close(results)
-			break
+LOOP:
+	for {
+		select {
+		case <-ctx.Done():
+			// request has been canceled, so no need to process the results
+			return nil, nil
+		case o := <-results:
+			if o.omitted {
+				tracing.Failure(span)
+				tracing.Error(span, errReadTooOld)
+				return nil, errReadTooOld
```

Comment:
well it is too old in the sense that the CRR outlived the originating "/render" request.

Reply:
It should also be noted that with the default cassandra-omit-read-timeout config, unless the client waits longer than 60seconds for a request to complete (not possible in hosted-metrics as the ingress-controller will timeout after 60seconds),  o.omitted will only ever be set for requests that are cancelled (due to timeout or other error)

Reply:
right. the reason i brought this up is because it seemed useful to separate these 2 cases in the error being logged in the trace, but actually it doesn't seem to matter much so this looks fine.